### DISCUSSION TITLE
MJr: a language heavily inspired by MarkovJunior

### DISCUSSION THREAD

**Discussion by kaya3** [mxgmn/MarkovJunior#55] [Show and tell]
I mentioned in a couple of other threads that I was working on a compiler. Well, here it is! The language is called MJr, in homage to MarkovJunior. The GitHub project for it is [here](https://github.com/kaya3/MJr).

There's a lot that isn't supported yet, but all of the fundamentals are implemented: `one`, `all`, `prl`, `sequence`, `markov` and limits. You can also specify conditions for rules: a neat example of this is the BasicSnake model from MarkovJunior, to which I added a condition so the program ends when the snake gets stuck. [Here's a link](https://kaya3.github.io/MJr/playground/#IyBUcmFuc2xhdGVkIGZyb20gTWF4aW0gR3VtaW4ncyAnQmFzaWNTbmFrZScgbW9kZWwKIyBodHRwczovL2dpdGh1Yi5jb20vbXhnbW4vTWFya292SnVuaW9yL2Jsb2IvbWFpbi9tb2RlbHMvQmFzaWNTbmFrZS54bWwKCmdyaWQgW0JXRFBHUl0KCnB1dCBbV10gYXQgb3JpZ2luCgphbGw6IFtbV0RdQkJdIC0+IFsuLkRdCgpvbmNlOiBbV0JEXSAtPiBbUEdSXQoKQGxpbWl0IDIKb25lOiBbUkJEXSAtPiBbR0dSXQoKQGxpbWl0IDE1Cm9uZTogW0RdIC0+IFtXXQoKbWFya292OgogICAgb25lOiBbUkJXXSAtPiBbR0dSXQogICAgYWxsOgogICAgICAgIFtSQkRdIC0+IFtHR1JdCiAgICAgICAgW1BHR10gLT4gW0RCUF0gaWYgY291bnQgW1JCRF0gPiAwCg==,JavaScript,1,1,31,31,-1) to that example in the interactive playground.

I want to say thanks a lot to Maxim for introducing me to this wonderful programming paradigm. I've had great fun with this project so far, and there's still so much left to do!

**Comment by mxgmn** [comment]
Cool! I'll link it from the README. Is there a way to browse examples in the playground?

**Reply by kaya3** [reply]
Not yet, but I might add something like that when there are more examples. I don't want to write too many example programs yet because it's still possible I might make some changes to the syntax.

**Reply by mxgmn** [reply]
Also, cool Pac-Man example!
